src/librustc/ty/constness.rs

```diff
@@ -68,14 +69,60 @@ impl<'tcx> TyCtxt<'tcx> {
 
 
 pub fn provide(providers: &mut Providers<'_>) {
-    /// only checks whether the function has a `const` modifier
+    fn is_const_intrinsic(tcx: TyCtxt<'_>, def_id: DefId) -> bool {
+        // Const evaluability whitelist is here to check evaluability at the
+        // top level beforehand.
+        match tcx.fn_sig(def_id).abi() {
+            Abi::RustIntrinsic |
+            Abi::PlatformIntrinsic => {
+                match &tcx.item_name(def_id).as_str()[..] {
```

Comment:
Could we change this to match on a `Symbol` instead?

Reply:
Also, @nnethercote has been trying to remove queries that are easy to compute to improve performance. I think this should just be a function?